### DISCUSSION TITLE
PorousFlowPermeabilityKozenyCarman - derivatives

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#24175] [Q&A Modules: General] [answered]
Hi, 
in `PorousFlowPermeabilityKozenyCarman.C`  is it possible that:
    ```
_dpermeability_qp_dvar[_qp][v] = _dporosity_qp_dvar[_qp][v] * _permeability_qp[_qp] *
                                     (_n / _porosity_qp[_qp] + _m / (1.0 - _porosity_qp[_qp]));
```
should be?
  ```
 _dpermeability_qp_dvar[_qp][v] = _dporosity_qp_dvar[_qp][v] * _permeability_qp[_qp] *
                                     (_n / _porosity_qp[_qp] - _m / (1.0 - _porosity_qp[_qp]));
```

and similarly we would have?
      ```
_dpermeability_qp_dgradvar[_qp][i][v] =
          _dporosity_qp_dgradvar[_qp][v](i) * _permeability_qp[_qp] *
          (_n / _porosity_qp[_qp] - _m / (1.0 - _porosity_qp[_qp]));
```

Sorry if this is wrong.
Javier

**Comment by WilkAndy** [comment, marked as answer]
It's great to have more pairs of eyes on this, @garciapintado .   I think the current code is correct.  By the chain rule, there is one negative from the negative exponent m (ie (1-phi)^(-m)) and one negative from the minus sign in front of phi.  Both of these multiply to give a positive sign.

If you're wondering about derivatives like these, you can probably find an input file in `porous_flow/test/tests/jacobian` that tests them.  You could modify the code, recompile, and run the test to see what you get.

A

**Reply by garciapintado** [reply]
Argh! You're right as always :-) Many thanks and for the info about the jacobian test
